core/jobs/batch_jobs/exp_validation_jobs.py

```diff
@@ -0,0 +1,311 @@
+        return invalid_state_names
+
+    def filter_exps_having_invalid_state_classifier_model_id(
+        self, states: dict[str, state_domain.State]) -> bool:
+        """Returns True if any state has classifier model id not None.
+
+        Args:
+            states: dict(str, State). A dict of states.
+
+        Returns:
+            bool. Returns True if any state has classifier model id not None.
+        """
+        return any(
+            state.classifier_model_id is not None for state in states.values()
+        )
+
+    def get_states_having_invalid_training_data(
+        self, exploration: exp_domain.Exploration) -> list[str]:
+        """Returns state names having non-empty training data.
+
+        Args:
+            exploration: Exploration. Exploration to be checked.
+
+        Returns:
+            list[str]. Returns state names having non-empty training data.
+        """
+        state_names: list[str] = []
+        states: dict[str, state_domain.State] = exploration.states
+        for state_name in states:
+            answer_groups: list[state_domain.AnswerGroup] = (
+                states[state_name].interaction.answer_groups)
+            for answer_group in answer_groups:
+                if len(answer_group.training_data) > 0:
+                    state_names.append(state_name)
+                    # The below break is necessary to avoid duplicated
+                    # state names in the list.
+                    break
+        return state_names
+
+    def filter_curated_explorations(
+        self, exp_and_opp_models: tuple[str, dict[str, list[object]]]):
+        """Returns whether the exp model is curated or not.
+
+        Args:
+            exp_and_opp_models: tuple. The pair of exp id and models.
+
+        Returns:
+            bool. Returns whether the exp model is curated or not.
+        """
+        exp_models_list = list(exp_and_opp_models[1]['exp_model'])
+        opp_models_list = list(exp_and_opp_models[1]['exp_opportinity_model'])
+        return len(exp_models_list) == 1 and len(opp_models_list) == 1
+
+    def get_exploration_from_models(
+        self, exp_and_opp_models: tuple[str, dict[str, list[object]]]):
+        """Returns the exploration domain object from the curated
+        exploration model.
+
+        Args:
+            exp_and_opp_models: tuple. The pair of exp and opportunity models.
+
+        Returns:
+            Exploration. The exploration domain object.
+        """
+        exp_model = list(exp_and_opp_models[1]['exp_model'])[0]
+        return exp_fetchers.get_exploration_from_model(exp_model)
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        """Returns PCollection of details of explorations which are invalid.
+
+        Returns:
+            PCollection[JobRunResult]. A PCollection of details of explorations
+            which are invalid.
+        """
+        all_explorations: beam.PCollection = (
```

Comment:
You don't need to add the types here. Only add types to places where MyPy checks fail for this file.

Reply:
ditto below

Reply:
Removed them